src/_pytest/pathlib.py

```diff
@@ -254,11 +254,10 @@ def maybe_delete_a_numbered_dir(path: Path) -> None:
     lock_path = None
     try:
         lock_path = create_cleanup_lock(path)
-        parent = path.parent
-
-        garbage = parent.joinpath(f"garbage-{uuid.uuid4()}")
-        path.rename(garbage)
-        rm_rf(garbage)
+        with tempfile.TemporaryDirectory(
```

Comment:
hmm I think this needs to be 0o755

https://github.com/python/cpython/blob/3d43f1dce3e9aaded38f9a2c73e3c66acf85505c/Lib/tempfile.py#L793

Reply:
https://bugs.python.org/issue26660

Reply:
ok I'm just doing it wrong - it works here:

```python
import tempfile
import pathlib


def main():
    with tempfile.TemporaryDirectory() as tmp:
        tmp_path = pathlib.Path(tmp)
        p = tmp_path / "ham" / "spam" / "eggs"
        p.mkdir(parents=True)
        p.touch(mode=0)
        for parent in p.parents:
            if parent == tmp_path:
                break
            parent.chmod(mode=0)

    assert not tmp_path.exists()


if __name__ == "__main__":
    main()
```